src/vos/vos_obj_index.c

```diff
@@ -321,6 +322,7 @@ vos_oi_find_alloc(struct vos_container *cont, daos_unit_oid_t oid,
 	if (log) {
 		vos_ilog_desc_cbs_init(&cbs, vos_cont2hdl(cont));
 		rc = ilog_open(vos_cont2umm(cont), &obj->vo_ilog, &cbs, dth == NULL, &loh);
+		D_ASSERTF(rc == -DER_NONEXIST, "Uncorrectable incarnation log corruption detected");
```

Comment:
rc != -DER_NONEXIST

Reply:
Ah, right, I fixed the other one when a unit test failed